### PR TITLE
Roi management

### PR DESCRIPTION
**Pull request by 14thibea**
This PR allows to use regions extracted by `clinicadl extract`.

To be merged after #119

### CODE REVIEW

File: docs/Train/Details.md [hunk 1]
```diff
@@ -125,3 +125,22 @@ where:
 
 - *p<sub>i</sub><sup>AD</sup>* is the probability of AD for patch *i*,
 - *bacc<sub>i</sub>* is the validation balanced accuracy for patch *i*.
+
+## Multi-cohort
+
+It is now possible to use fonctions of ClinicaDL on several datasets at the same time.
```
Comment (mdiazmel) [inline docs/Train/Details.md]:
```suggestion
Starting from version 0.2.1, it is possible to use functions of ClinicaDL on several datasets at the same time.
```

File: docs/Train/Details.md [hunk 2]
```diff
@@ -125,3 +125,22 @@ where:
...
+In this case, the `multi-cohort` flag must be given, and the `caps_directory` and the `tsv_path`
+must be TSV files.
```
Comment (mdiazmel) [inline docs/Train/Details.md]:
```suggestion
correspond to TSV files.
```

File: docs/Train/Details.md [hunk 3]
```diff
@@ -125,3 +125,22 @@ where:
...
+
+The `caps_directory` variable must lead to a TSV file with two columns:
+- `cohort` the name of the cohort (must correspond to the values in `tsv_path`),
+- `path` the path to the corresponding [CAPS](https://aramislab.paris.inria.fr/clinica/docs/public/latest/CAPS/Introduction/) hierarchy.
+
+The `tsv_path` variable must lead to a TSV file with two columns:
```
Comment (mdiazmel) [inline docs/Train/Details.md]:
```suggestion
The `tsv_path` argument points to a TSV file with two columns:
```

File: docs/Train/ROI.md [hunk 4]
```diff
@@ -1,13 +1,14 @@
 # `train roi` - Train deep learning networks using predefined regions of interest (ROI)
 
-This option allows training a network on two regions of interest (ROI).
-ROI inputs correspond to two patches of size 50x50x50 voxels manually centered on each hippocampus.
-This manual centering has only been done for the `t1-linear` pipeline.
+This option allows training a network on a set of regions of interest (ROI) defined by
```
Comment (mdiazmel) [inline docs/Train/ROI.md]:
```suggestion
This option allows training a network using a set of regions of interest (ROI) defined by 
```

File: docs/Train/ROI.md [hunk 5]
```diff
@@ -116,6 +141,14 @@ where mandatory arguments are:
 
 The options specific to this task are the following:
 
+- `--roi_list` (list of str) includes the names of the regions wanted. Each region must correspond
```
Comment (mdiazmel) [inline docs/Train/ROI.md]:
```suggestion
- `--roi_list` (list of str) includes the names of the targeted regions. Each region corresponds
```

File: docs/Train/ROI.md [hunk 6]
```diff
@@ -116,6 +141,14 @@ where mandatory arguments are:
...
+to a mask defined in `caps_directory`. See the [dedicated section](./ROI.md#definition-of-masks) for more information.
+Default will extract two patches centered on the hippocampi (available for `t1-linear` preprocessing only).
```
Comment (mdiazmel) [inline docs/Train/ROI.md]:
```suggestion
Default behavior will extract two patches centered on the hippocampi (available for `t1-linear` preprocessing only).
```

**Review by mdiazmel — COMMENTED** [review]
Hi @14thibea ,
I finish the review for this PR. I had the impression that some of the changes where also in a previous PR. The comments concerns mostly the documentation.
For testing`train_custom_roi`, I suppose that it uses the dataset available in the server... isn't?

**Comment by 14thibea** [comment]
Hi @mdiazmel ,

Thanks for the review! 

Yes and I see that there are conflicts too... I guess that these two issues are related. A rebase will be quite long to do according to the number of commits is it OK for you if I merge dev in this branch?

Indeed this new test uses the dataset in the server, so no need to import anything, it just generates random masks and checks that it works using them.

**Comment by mdiazmel** [comment]
> Yes and I see that there are conflicts too... I guess that these two issues are related. A rebase will be quite long to do according to the number of commits is it OK for you if I merge dev in this branch?

No problem, go ahead! We will see later what strategy we should use to add this contribution to the dev branch (maybe a `squash and merge` should produce a clearer commit history at the end).

**Comment by mdiazmel** [comment]
I'll wait  you to fix the conflicts and solve the comments in the revision before merging! Don't hesitate to poke me when it's ready!

**Pull request merged**